### PR TITLE
Reenable pytorch 4x4solver

### PR DESCRIPTION
**Pull request by MarJMue**

### CODE REVIEW

File: src/elli/solver4x4.py [hunk 1]
```diff
@@ -71,7 +76,30 @@ def calculate_propagation(
         """
         mats = 1j * thickness * np.einsum("nij,n->nij", delta, 2 * sc.pi / lbda)
 
-        propagator = np.asarray([scipy_expm(mat) for mat in mats])
+        propagator = scipy_expm(mats)
```
Comment (domna) [inline src/elli/solver4x4.py]: I suggest to just have one expm class and select the appropriate backend automatically based on availability. I think users might not know which propagator to use and we should not confuse them with too many details. We could give the class a `backend` property from which another backend can be selected. So users could select `backend=torch/scipy/automatic`.

Comment (domna) [reply src/elli/solver4x4.py]: We can just build something like this in the init:
```python
def __init__(self, backend: Literal["torch", "scipy", "automatic"] = "automatic"):
  backends = {
    "torch": lambda mats: torch.linalg.matrix_exp(torch.from_numpy(mats)).numpy(),
    "scipy": lambda mats: scipy_expm(mats)
  }
  if backend == "torch" and not TORCH_AVAILABLE:
    raise ... or log
  if backend not in backends:
    backend = "torch"
  self.expm = backends[backend]
```

and then just use `self.expm` in the code.

Comment (MarJMue) [reply src/elli/solver4x4.py]: Completely agree. Defaults are important and if someone happens to have PyTorch installed, they will have a better experience.

**Review by domna — APPROVED** [review]
LGTM

**Pull request merged**